MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,230 @@
+        Mapbox.getInstance(this, getString(R.string.access_token));
+
+        setContentView(R.layout.activity_extrusion_rotation);
+
+        mapView = (MapView) findViewById(R.id.mapView);
+        mapView.onCreate(savedInstanceState);
+        mapView.getMapAsync(new OnMapReadyCallback() {
+            @Override
+            public void onMapReady(@NonNull final MapboxMap map) {
+                mapboxMap = map;
+                setupBuildings();
+                setupLocationPlugin();
+            }
+        });
+
+        //initialize sensors
+        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+        gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+
+        sensorManager.registerListener(this, gyro , 200);
+        sensorManager.registerListener(this, magnetic , 200);
+    }
+
+    private void setupBuildings() {
+        FillExtrusionLayer fillExtrusionLayer = new FillExtrusionLayer("3d-buildings", "composite");
+        fillExtrusionLayer.setSourceLayer("building");
+        fillExtrusionLayer.setFilter(eq("extrude", "true"));
+        fillExtrusionLayer.setMinZoom(15);
+        fillExtrusionLayer.setProperties(
+                fillExtrusionColor(Color.LTGRAY),
+                fillExtrusionHeight(Function.property("height", new IdentityStops<Float>())),
+                fillExtrusionBase(Function.property("min_height", new IdentityStops<Float>())),
+                fillExtrusionOpacity(0.9f)
+        );
+        mapboxMap.addLayer(fillExtrusionLayer);
+    }
+
+    private void setupLocationPlugin(){
+//        locationEngine = new LocationSource(this);
+//        locationLayerPlugin = new LocationLayerPlugin(mapView, mapboxMap, locationEngine);
+//        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+//            return;
+//        }
+//        locationLayerPlugin.setLocationLayerEnabled(COMPASS);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_building, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (light != null) {
+            int id = item.getItemId();
+            if (id == R.id.menu_action_anchor) {
+                isMapAnchorLight = !isMapAnchorLight;
+                light.setAnchor(isMapAnchorLight ? Property.ANCHOR_MAP : Property.ANCHOR_VIEWPORT);
+            } else if (id == R.id.menu_action_intensity) {
+                isLowIntensityLight = !isLowIntensityLight;
+                light.setIntensity(isLowIntensityLight ? 0.35f : 1.0f);
+            } else if (id == android.R.id.home) {
```

Comment:
These menu option items don't seem to work for me. Nothing about the light on the buildings seems to change.

 Do they work for you @electrostat ? Here's what it looks like on my device:

![ezgif com-resize](https://user-images.githubusercontent.com/4394910/28850111-8c51be48-76cf-11e7-9dd3-952029609d07.gif)